targets/REV3-SGM/main.cpp

```diff
@@ -1,30 +1,146 @@
- * basic echo functionality where the uart will write back whatever the user
- * enters.
+ * This target collects and transmits data from multiple strain gauges to the CANopen network via the SGM board.
+ * The target reads ADC values, performs slight processing, and sends the data to the CANopen network.
  */
 
+#include <EVT/dev/MCUTimer.hpp>
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/CANOpenMacros.hpp>
+#include <EVT/io/CANopen.hpp>
 #include <EVT/io/UART.hpp>
 #include <EVT/io/pin.hpp>
 #include <EVT/manager.hpp>
+#include <EVT/utils/log.hpp>
+#include <EVT/utils/time.hpp>
 #include <SGM.hpp>
 
 namespace IO = EVT::core::IO;
+namespace log = EVT::core::log;
+namespace time = EVT::core::time;
+namespace DEV = EVT::core::DEV;
+
+/**
+ * Note, this conversion function is just to show off how a conversion
+ * function will look like. In actuality, this will be determined through
+ * testing/reading the strain gauge datasheet.
+ */
+uint32_t convert(uint32_t voltage) {
+    return voltage / 2;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// EVT-core CAN callback and CAN setup. This will include logic to set
+// aside CANopen messages into a specific queue
+///////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Interrupt handler to get CAN messages. A function pointer to this function
+ * will be passed to the EVT-core CAN interface which will in turn call this
+ * function each time a new CAN message comes in.
+ *
+ * NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
+ * assumed to be intended to be passed as a CANopen message.
+ *
+ * @param message[in] The passed in CAN message that was read.
+ */
+
+// create a can interrupt handler
+void canInterrupt(IO::CANMessage& message, void* priv) {
+    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage>* queue =
+        (EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage>*) priv;
+
+    // Log raw received data
+    log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Got RAW message from %X of length %d with data: ", message.getId(), message.getDataLength());
+
+    uint8_t* data = message.getPayload();
+    for (int i = 0; i < message.getDataLength(); i++) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "%X ", *data);
+        data++;
+    }
+
+    if (queue != nullptr)
+        queue->append(message);
+}
 
 int main() {
     // Initialize system
     EVT::core::platform::init();
 
     // Setup UART
     IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    log::LOGGER.setUART(&uart);
+
+    // Setup pins for Strain Gauges
+    IO::ADC& adc0 = IO::getADC<IO::Pin::PC_0>();// Strain Gauge D
+    IO::ADC& adc1 = IO::getADC<IO::Pin::PC_1>();// Strain Gauge C
+    IO::ADC& adc2 = IO::getADC<IO::Pin::PC_2>();// Strain Gauge B
+    IO::ADC& adc3 = IO::getADC<IO::Pin::PC_3>();// Strain Gauge A
```

Comment:
I would order these in alphabetical order and rename the variables to be things like `adcA`, so the software corresponds more clearly to the hardware ordering